ivy/functional/backends/tensorflow/experimental/manipulation.py

```diff
@@ -271,3 +273,89 @@ def concat_from_sequence(
     elif new_axis == 1:
         ret = tf.stack(input_sequence, axis=axis)
         return ret
+        
+
+def _check_tuple(t):
+    if isinstance(t, tuple):
+        if len(t) == 1 and not isinstance(t[0], tuple):
+            return t[0], True
+        else:
+            has_multiple_values = False
+            size = 0
+            for elem in t:
+                elem_val, elem_has_multiple_values = _check_tuple(elem)
+                if elem_has_multiple_values:
+                    has_multiple_values = True
+                size += elem_val
+            return size, has_multiple_values
+    else:
+        return 1, False
+
+
+def _check_dimension(tensor, padding, mode):
+    if mode == 'reflect':
+
+        for i in range(tensor.shape.rank):
+            if padding[i][0] > tensor.shape[i] - 1 or padding[i][1] > tensor.shape[i] - 1:
+                return False
+            return True
+
+    elif mode == 'symmetric':
+
+        for i in range(tensor.shape.rank):
+            if padding[i][0] > tensor.shape[i] or padding[i][1] > tensor.shape[i]:
+                return False
+        return True
+
+
+def _check(*args, **kwargs):
+
+    mode = kwargs['mode']
+    if mode in ["linear_ramp",
+                    "maximum",
+                    "mean",
+                    "median",
+                    "minimum",
+                    "edge",
+                    "wrap",
+                    "empty",
+    ]:
+        return False
+    else:
+        if mode == 'constant':
+            c = kwargs['constant_values']
+            val, cond = _check_tuple(c)
+            if cond is False:
+                return False
+            else:
+                kwargs['constant_values'] = val
+                return True
+        else:
+            pad = args[1]
+            inp = args[0]
+            if isinstance(pad, tuple):
+                if _check_dimension(inp, pad, mode):
+                    return True
+                else:
+                    return False
+            elif isinstance(pad, int):
+                return False
```

Comment:
Could you explain why you are making these checks for `constant_values` and `pad`? I was expecting that the only thing to check for tensorflow would be the `mode`.

Reply:
In the documentation of the Tensorflow, we also need to check the dimensions of specific modes (SYMMETRIC AND REFLECT) and if the dimensions of the pad meet the criteria that are given in the Tensorflow documentation then we can use native implementation. if it does not meet the criteria then it raises an error so that's why I added this logic that it should call the compositional implementation of the pad when the modes are symmetric or reflect and if it does not meet the criteria.  Here is the documentation of the tf.pad that explains this condition: [Click Here](https://www.tensorflow.org/api_docs/python/tf/pad)

Reply:
Right, tensorflow always has to introduce weird checks and conditions 😅

Reply:
yes, it has weird conditions 😅